rustls-symcrypt/src/signer.rs

```diff
@@ -1,22 +1,321 @@
+                    Err(e) => Err(Error::General(format!("failed to sign message: {}", e))),
+                }
+            }
+            SignatureScheme::RSA_PKCS1_SHA512 => {
+                let hashed_message = sha512(message);
+                match self.key.pkcs1_sign(&hashed_message, HashAlgorithm::Sha512) {
+                    Ok(signature) => Ok(signature),
+                    Err(e) => Err(Error::General(format!("failed to sign message: {}", e))),
+                }
+            }
+            SignatureScheme::RSA_PSS_SHA256 => {
+                let hashed_message = sha256(message);
+                match self.key.pss_sign(&hashed_message, HashAlgorithm::Sha256, 32) {
+                    Ok(signature) => Ok(signature),
+                    Err(e) => Err(Error::General(format!("failed to sign message: {}", e))),
+                }
+            }
+            SignatureScheme::RSA_PSS_SHA384 => {
+                let hashed_message = sha384(message);
+                match self.key.pss_sign(&hashed_message, HashAlgorithm::Sha384, 48) {
+                    Ok(signature) => Ok(signature),
+                    Err(e) => Err(Error::General(format!("failed to sign message: {}", e))),
+                }
+            }
+            SignatureScheme::RSA_PSS_SHA512 => {
+                let hashed_message = sha512(message);
+                match self.key.pss_sign(&hashed_message, HashAlgorithm::Sha512, 64) {
+                    Ok(signature) => Ok(signature),
+                    Err(e) => Err(Error::General(format!("failed to sign message: {}", e))),
+                }
+            }
+            _ => {
+                return Err(Error::General("unsupported RSA-PKCS1 signature scheme".into()));
+            }
+        }
+    }
+
+    fn scheme(&self) -> SignatureScheme {
+        self.scheme
+    }
+}
+
+static ALL_ECDSA_SCHEMES: &[SignatureScheme] = &[
+    SignatureScheme::ECDSA_NISTP256_SHA256,
+    SignatureScheme::ECDSA_NISTP384_SHA384,
+    SignatureScheme::ECDSA_NISTP521_SHA512,
+];
+
+#[derive(Debug)]
+pub struct EcdsaSigningKey {
+    key: Arc<EcKey>, 
+}
+
+
+impl EcdsaSigningKey {
+    pub fn new(der: &PrivateKeyDer<'_>) -> Result<Self, Error> {
+        let key = match der {
+            PrivateKeyDer::Pkcs1(pkcs1) => {
+                // Get the DER-encoded private key blob
+                let private_key_blob = pkcs1.secret_pkcs1_der();
+
+                // Determine the curve type for PKCS#1
+                let curve_type = match private_key_blob {
+                    blob if blob.starts_with(&[0x30, 0x81]) => CurveType::NistP256,
+                    blob if blob.starts_with(&[0x30, 0x82]) => CurveType::NistP384,
+                    blob if blob.starts_with(&[0x30, 0x83]) => CurveType::NistP521,
+                    blob if blob.starts_with(&[0x30, 0x84]) => CurveType::Curve25519,
+                    _ => return Err(Error::General("Unsupported curve type".into())),
```

Comment:
Where is this coming from??

You first need to attempt to parse the ECC private key. Then, do the curve comparison.